### ISSUE TITLE
Investigate performance in TensorFlow

### ISSUE THREAD

**Issue by gmastrapas** [jina-ai/finetuner#300]
When fine-tuning a ResNet for fashion image search, using finetuner and TensorFlow, we observed that the training loss was not decreasing.

**Comment by tadejsv** [comment]
Closed due to dropping of tf and paddle in near future

**Comment by guenthermi** [comment]
We tested the keras ResNet50 model and the PyTorch ResNet50 model on the [Totally Looks Like dataset](https://sites.google.com/view/totally-looks-like-dataset). The problem probably comes from a wrong image normalization. By using the function `tensorflow.keras.applications.resnet50.preprocess_input` for normalization, keras achieves results which are comparable to the results of the PyTorch and Paddle ResNet50 models:

PyTorch:
{"average_precision": 0.19490375491652837, "dcg_at_k": 0.2769598038571193, "f1_score_at_k": 0.04160734061066286, "hit_at_k": 0.4368770764119601, "ndcg_at_k": 0.2769598038571193, "precision_at_k": 0.021843853820598006, "r_precision": 0.19490375491652837, "recall_at_k": 0.4368770764119601, "reciprocal_rank": 0.19490375491652837}

Paddle:
{"average_precision": 0.17202600897103484, "dcg_at_k": 0.2509920683958687, "f1_score_at_k": 0.038680588514475554, "hit_at_k": 0.4061461794019934, "ndcg_at_k": 0.2509920683958687, "precision_at_k": 0.02030730897009967, "r_precision": 0.17202600897103484, "recall_at_k": 0.4061461794019934, "reciprocal_rank": 0.17202600897103484}

Keras:
{"average_precision": 0.22903498850644843, "dcg_at_k": 0.3338376748490199, "f1_score_at_k": 0.04872646733111849, "hit_at_k": 0.5116279069767442, "ndcg_at_k": 0.3338376748490199, "precision_at_k": 0.02558139534883721, "r_precision": 0.22903498850644843, "recall_at_k": 0.5116279069767442, "reciprocal_rank": 0.22903498850644843}
